inc/Engine/Optimization/RUCSS/Warmup/ResourceFetcher.php

```diff
@@ -0,0 +1,169 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Engine\Optimization\AbstractOptimization;
+use WP_Rocket\Engine\Optimization\AssetsLocalCache;
+use WP_Rocket\Logger\Logger;
+
+class ResourceFetcher extends AbstractOptimization {
+
+	/**
+	 * Resources array.
+	 *
+	 * @var array
+	 */
+	private $resources = [];
+
+	/**
+	 * Assets local cache instance
+	 *
+	 * @var AssetsLocalCache
+	 */
+	private $local_cache;
+
+	/**
+	 * Resource fetcher process instance.
+	 *
+	 * @var ResourceFetcherProcess
+	 */
+	private $process;
+
+	/**
+	 * Resource constructor.
+	 *
+	 * @param AssetsLocalCache       $local_cache Local cache instance.
+	 * @param ResourceFetcherProcess $process     Resource fetcher process instance.
+	 */
+	public function __construct( AssetsLocalCache $local_cache, ResourceFetcherProcess $process ) {
+		$this->local_cache = $local_cache;
+		$this->process     = $process;
+	}
+
+	/**
+	 * Handle Collecting resources and save them into the DB
+	 *
+	 * @param string $html Page HTML.
+	 */
+	public function handle( $html ) {
+		// Grab resources from page HTML.
+		$this->find_styles( $html );
+		$this->find_scripts( $html );
+
+		// Send resources to the background process to be saved into DB.
+		$this->process->push_to_queue( $this->resources );
```

Comment:
In this case, why was the choice of pushing the full array to the background process made, instead of looping over the array to push every item of the array individually?

Reply:
Ok, I was between two choices:-
1. Send array of resources to be sent to saas warmup bulky =>will take some time and may timeout
2. Send item per item to watmup process so thay can be handled in parallel

what do you think ?

Reply:
Since we're sending to the background process, I think it makes more sense to go with option 2, as it's the whole idea of the background process: handle multiple, small items one by one instead of a big one

Reply:
But it also means that you will have to handle passing data to the next BG process differently, as in the `task()` method you won't have everything available, only the current item.